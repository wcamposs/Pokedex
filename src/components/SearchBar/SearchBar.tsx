import React from 'react';
import { StyleSheet, TextInput, View } from 'react-native';
import { Feather } from '@expo/vector-icons';

import colors from '../../utils/colors';

function SearchBar(props: any) {
	const { onBlur, onChangeText, value } = props;

	function handleOnChangeText(text = '') {
		onChangeText(text);
	}

	function handleOnCleanText() {
		if (!value.length) {
			return;
		}

		onChangeText('');
	}

	return (
		<View style={styles.container}>
			<TextInput
				autoCapitalize="none"
				autoCompleteType="off"
				onChangeText={handleOnChangeText}
				placeholder="Search a pokemon..."
				selectTextOnFocus
				style={styles.input}
				value={value}
				returnKeyType="search"
				onBlur={onBlur}
			/>
			{!value.length ? null : (
				<Feather
					color={colors.darkBlue}
					name="x"
					onPress={handleOnCleanText}
					size={20}
					type="light"
					containerStyle={styles.icon}
				/>
			)}
		</View>
	);
}

const styles = StyleSheet.create({
	container: {
		alignItems: 'center',
		backgroundColor: colors.white,
		borderColor: colors.grey,
		borderRadius: 12,
		borderWidth: 0.5,
		flexDirection: 'row',
		paddingHorizontal: 12,
		marginTop: 12,
		marginHorizontal: 12,
		height: 42,
	},
	icon: {
		padding: 4,
	},
	input: {
		flex: 1,
		paddingHorizontal: 12,
		paddingVertical: 12,
	},
});

export default SearchBar;
