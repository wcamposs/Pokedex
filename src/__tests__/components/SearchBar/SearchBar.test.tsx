import React from 'react';
import renderer from 'react-test-renderer';

import SearchBar from '../../../components/SearchBar/SearchBar';

jest.useFakeTimers();

describe('<SearchBar />', () => {
	it('renders correctly without value', () => {
		const onBlur = jest.fn();
		const onChangeText = jest.fn();

		const tree = renderer
			.create(
				<SearchBar
					onBlur={onBlur}
					onChangeText={onChangeText}
					value=""
				/>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders correctly with an value', () => {
		const onBlur = jest.fn();
		const onChangeText = jest.fn();

		const tree = renderer
			.create(
				<SearchBar
					onBlur={onBlur}
					onChangeText={onChangeText}
					value="Pikachu"
				/>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
