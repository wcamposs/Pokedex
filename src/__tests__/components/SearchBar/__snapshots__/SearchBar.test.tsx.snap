// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> renders correctly with an value 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderColor": "#e6e6f0",
      "borderRadius": 12,
      "borderWidth": 0.5,
      "flexDirection": "row",
      "height": 42,
      "marginHorizontal": 12,
      "marginTop": 12,
      "paddingHorizontal": 12,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCompleteType="off"
    onBlur={[MockFunction]}
    onChangeText={[Function]}
    placeholder="Search a pokemon..."
    rejectResponderTermination={true}
    returnKeyType="search"
    selectTextOnFocus={true}
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 12,
      }
    }
    underlineColorAndroid="transparent"
    value="Pikachu"
  />
  <Text />
</View>
`;

exports[`<SearchBar /> renders correctly without value 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderColor": "#e6e6f0",
      "borderRadius": 12,
      "borderWidth": 0.5,
      "flexDirection": "row",
      "height": 42,
      "marginHorizontal": 12,
      "marginTop": 12,
      "paddingHorizontal": 12,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCompleteType="off"
    onBlur={[MockFunction]}
    onChangeText={[Function]}
    placeholder="Search a pokemon..."
    rejectResponderTermination={true}
    returnKeyType="search"
    selectTextOnFocus={true}
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 12,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
